Backport main to net7.0

Redth opened the pull request:
Backport main to net7.0

PureWeen committed:
Scoping doc (#4679)

* Scoping doc

* Update Scoping.md

PureWeen committed:
Create Window Specific Orientation Extensions (#7338)

* Create Window Specific Orientation Extensions

* Update src/Core/src/Platform/Tizen/WindowExtensions.cs

Co-authored-by: Kangho Hur <kangho.hur@samsung.com>

* Update src/Core/src/Platform/Tizen/WindowExtensions.cs

Co-authored-by: Kangho Hur <kangho.hur@samsung.com>

Co-authored-by: Kangho Hur <kangho.hur@samsung.com>

rmarinho committed:
[iOS] Fix default naming on iOS cells (#7302)

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/xharness build 20220520.1 (#7414)

Microsoft.DotNet.XHarness.CLI , Microsoft.DotNet.XHarness.TestRunners.Common , Microsoft.DotNet.XHarness.TestRunners.Xunit
 From Version 1.0.0-prerelease.22261.2 -> To Version 1.0.0-prerelease.22270.1

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

jsuarezruiz committed:
Fix crash removing items in ListView (#7411)

StephaneDelcroix committed:
[X] Add compiled converters (#7245)

* [X] Add compiled converters

- Add compiled converters for Flex enums
- and for FlexBasis
- and for for FlowDirection
- Also moves Flowdirection converter attribute from the property to the
  Type, and fix the Design assembly accordingly

- fixes #6547

* Update src/Core/src/Layouts/FlexEnums.cs

Co-authored-by: campersau <buchholz.bastian@googlemail.com>

Co-authored-by: campersau <buchholz.bastian@googlemail.com>

hartez committed:
Allow the image handler to measure more accurately WRT aspect ratio (#7389)

Fix for issue 2000 on Android

rachelkang committed:
Fix RTL in AbsoluteLayout (#7309)

* Add AbsoluteLayoutManager RTL tests

* First attempt at fix

* Fix up tests and RTL math

* Remove unnecessary using

* Remove unneeded and wrong code

hartez committed:
Ensure that Grid is propagating the BindingContext to the row/col definitions (#7301)

Fixes #4609

jsuarezruiz committed:
Added ImageButton sample using VisualStateManager (#7296)

jsuarezruiz committed:
Fix grouped ListView ScrollTo crash on Windows (#7413)

mattleibow committed:
Remove the padding around checkboxes (#7379)

* Remove the padding around checkboxes

* Fix Windows

* Just centre the image and let the OS do the rest

beeradmoore committed:
Brings Android NamedSize font for Header down from 96 to 14 to match what it is in Xamarin.Forms (#7433)

rmarinho committed:
Revert "Fix crash removing items in ListView (#7411)" (#7418)

This reverts commit 4b64fd9d70b2b1e303dc3c189c1db282e7b1392f.

mattleibow committed:
Don't clear old images. Fixes #6625 (#7348)

* Don't clear old images. Fixes #6625

* omg

* Unify the image tests

* using

* test

* SetImageDrawable

jsuarezruiz committed:
Fixed crash using Path in ControlTemplate ContentView (#7412)

jsuarezruiz committed:
Fixed in Windows Shadows (#7336)

MackinnonBuck committed:
Fix BlazorWebView disposal on Android (#7349)

PureWeen committed:
Fix WinUI Modal to work with lack of root panel (#7452)

* Fix WinUI Modal to work with new Root

* - fix modal page button name

* Update ModalPage.xaml

* Update ModalPage.xaml.cs

Eilon committed:
Update bug template to list 6.0 GA (#7480)

Fixes #7474

jsuarezruiz committed:
Fix issues clipping content in Border on Android (#7467)

dependabot[bot] committed:
Bump Xamarin.AndroidX.Migration from 1.0.8 to 1.0.10 (#7469)

Bumps Xamarin.AndroidX.Migration from 1.0.8 to 1.0.10.

---
updated-dependencies:
- dependency-name: Xamarin.AndroidX.Migration
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump Microsoft.CodeAnalysis.NetAnalyzers (#7440)

Bumps [Microsoft.CodeAnalysis.NetAnalyzers](https://github.com/dotnet/roslyn-analyzers) from 7.0.0-preview1.22229.1 to 7.0.0-preview1.22274.2.
- [Release notes](https://github.com/dotnet/roslyn-analyzers/releases)
- [Changelog](https://github.com/dotnet/roslyn-analyzers/blob/main/PostReleaseActivities.md)
- [Commits](https://github.com/dotnet/roslyn-analyzers/commits)

---
updated-dependencies:
- dependency-name: Microsoft.CodeAnalysis.NetAnalyzers
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

drasticactions committed:
Remove Unsupported "!!" syntax from BlazorWebView (#7484)

PureWeen committed:
Fix swapping shell for navpage (#7415)

jfversluis committed:
Use .NET 6 built-in dotnet format tool (#6371)

rmarinho committed:
Update dotnet-format-daily.yml

rmarinho committed:
Update dotnet-format-daily.yml

rmarinho committed:
Update dotnet-format-daily.yml

rmarinho committed:
Update dotnet-format-daily.yml

rmarinho committed:
Update dotnet-format-daily.yml

rmarinho committed:
Fix Format (#7500)

github-actions[bot] committed:
[create-pull-request] automated change (#7506)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

rmarinho committed:
[Handlers] Make sure to update Content on handlers when the ControlTemplate changes (#7235)

* [Handlers] Make sure to update Content on handlers when the ControlTemplate changes

* [DeviceTests] Fix tests NRE

* [DeviceTests] Add device tests for ControlTemplate updates on ContentView

* Update handler value after calling base ApplyTemplate

* [Tests] Fix get childcount when there's no child

PureWeen committed:
Resolving TableView/ListView issues (#7485)

* Resolving TableView/ListView issues

* - remove dispose call

* Update ListViewAdapter.cs

* - add more tests

mattleibow committed:
Set GenerateDocumentationFile=true for Compatibility and Xaml and pack a special docs artifact (#7511)

github-actions[bot] committed:
[create-pull-request] automated change (#7517)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

PureWeen committed:
If the FlyoutBehavior is flyout always show toolbar (#7544)

mattleibow committed:
Merge the PublicAPI.*.txt from 6.0.3xx

jsuarezruiz committed:
Updated templates (#7439) Fixes #7424

jsuarezruiz committed:
Avoid crash with invalid casting using RadioButtonGroup.GroupName with a View (#7438) Fixes #7347

FriedrichRehren committed:
Just a small styling error I noticed just now (#7519)

dependabot[bot] committed:
Bump System.Memory from 4.5.4 to 4.5.5 (#7533)

Bumps [System.Memory](https://github.com/dotnet/corefx) from 4.5.4 to 4.5.5.
- [Release notes](https://github.com/dotnet/corefx/releases)
- [Commits](https://github.com/dotnet/corefx/commits)

---
updated-dependencies:
- dependency-name: System.Memory
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/xharness build 20220526.1 (#7605)

Microsoft.DotNet.XHarness.CLI , Microsoft.DotNet.XHarness.TestRunners.Common , Microsoft.DotNet.XHarness.TestRunners.Xunit
 From Version 1.0.0-prerelease.22270.1 -> To Version 1.0.0-prerelease.22276.1

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump Microsoft.CodeAnalysis.NetAnalyzers (#7606)

Bumps [Microsoft.CodeAnalysis.NetAnalyzers](https://github.com/dotnet/roslyn-analyzers) from 7.0.0-preview1.22274.2 to 7.0.0-preview1.22277.2.
- [Release notes](https://github.com/dotnet/roslyn-analyzers/releases)
- [Changelog](https://github.com/dotnet/roslyn-analyzers/blob/main/PostReleaseActivities.md)
- [Commits](https://github.com/dotnet/roslyn-analyzers/commits)

---
updated-dependencies:
- dependency-name: Microsoft.CodeAnalysis.NetAnalyzers
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

github-actions[bot] committed:
[create-pull-request] automated change (#7617)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

rmarinho committed:
[iOS] Fix cell size changed on CollectionView (#7447)

* [VisualElement] Bubble MeasureInvalidated from children to the parents

* [iOS] Remove not used code

* [iOS] CollectionView is not reloading its cell size when content change, try use ReloadItems instead InvalidateLayout

* [Core] Fix Invalidation just for new Layouts

jsuarezruiz committed:
Fix iOS ActivityIndicator with initial IsRunning value to False (#7609)

github-actions[bot] committed:
[create-pull-request] automated change (#7648)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

rmarinho committed:
[CollectionView] Make sure we map MapSelectionMode and SelectedITems (#7658)

pictos committed:
Updated the VS-target (#7649)

* Updated the VS-target

* Added a target that works for me

mattleibow committed:
Revert a type move for net6 (#7642)

PureWeen committed:
Hide non visible shell parts from VO (#7512)

* Hide non visible shell parts from VO

* - wrong order

* Update ShellFlyoutRenderer.cs

jsuarezruiz committed:
Fix crash setting zero or negative values in Border size (#7665)

mattleibow committed:
Merge the PublicAPI checker into main (#7339)

jsuarezruiz committed:
[Android] Implement quit Application (#7633)

* Implement quit App on Android

* Removed unnecessary validation

github-actions[bot] committed:
[create-pull-request] automated change (#7677)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

jsuarezruiz committed:
Fix Android Button CornerRadius issue (#7215)

Co-authored-by: Matthew Leibowitz <mattleibow@live.com>

PureWeen committed:
Propagate semantics to CV Item container (#7539)

Co-authored-by: Matthew Leibowitz <mattleibow@live.com>

mattleibow committed:
This is more correct (#7685)

This corrects a mistake in the #7339

dependabot[bot] committed:
Bump Microsoft.CodeAnalysis.NetAnalyzers (#7696)

Bumps [Microsoft.CodeAnalysis.NetAnalyzers](https://github.com/dotnet/roslyn-analyzers) from 7.0.0-preview1.22277.2 to 7.0.0-preview1.22301.1.
- [Release notes](https://github.com/dotnet/roslyn-analyzers/releases)
- [Changelog](https://github.com/dotnet/roslyn-analyzers/blob/main/PostReleaseActivities.md)
- [Commits](https://github.com/dotnet/roslyn-analyzers/commits)

---
updated-dependencies:
- dependency-name: Microsoft.CodeAnalysis.NetAnalyzers
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

github-actions[bot] committed:
[create-pull-request] automated change (#7714)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump Xamarin.UITest from 3.2.7 to 3.2.8 (#7697)

Bumps Xamarin.UITest from 3.2.7 to 3.2.8.

---
updated-dependencies:
- dependency-name: Xamarin.UITest
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from xamarin/xamarin-android (#7630)

* Update dependencies from https://github.com/xamarin/xamarin-android build 6.0.4xx-00dcdf0b16a68927250e72c9b4de61cdf368e4ab-1

Microsoft.Android.Sdk.Windows
 From Version 32.0.301 -> To Version 32.0.403

Dependency coherency updates

Microsoft.Dotnet.Sdk.Internal,Microsoft.NETCore.App.Ref
 From Version 6.0.301-rtm.22254.17 -> To Version 6.0.400-preview.22270.3 (parent: Microsoft.Android.Sdk.Windows

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20220523.14

Microsoft.tvOS.Sdk , Microsoft.macOS.Sdk , Microsoft.MacCatalyst.Sdk , Microsoft.iOS.Sdk
 From Version 15.4.303 -> To Version 15.4.403

* [Tizen] Update Samsung Tizen SDK Version for 6.0.400 band (#7640)

* Tizen is 6.0.400 band?

Context: https://www.nuget.org/packages/Samsung.NET.Sdk.Tizen.Manifest-6.0.400

* xamarin-macios is 6.0.400 band?

* Hardcode more version bands

* Revert "[Tizen] Update Samsung Tizen SDK Version for 6.0.400 band (#7640)"

This reverts commit 9b769f3bf1ed5cc8122e17b9f28fc5fe37077572.

* Tizen is 6.0.300 version band

* Revert "Revert "[Tizen] Update Samsung Tizen SDK Version for 6.0.400 band (#7640)""

This reverts commit 18f29ca47bea0647553e13ffd51d30ff4b9ae21f.

* Revert "Tizen is 6.0.300 version band"

This reverts commit 8ecbbf2a50d0f9868d6c0320dffdea761192aa63.

* Update dependencies from https://github.com/xamarin/xamarin-android build 6.0.4xx-387a92a0a1cafe22e7b16cfd6c4bc79bd824ac5c-1

Microsoft.Android.Sdk.Windows
 From Version 32.0.301 -> To Version 32.0.414

Dependency coherency updates

Microsoft.Dotnet.Sdk.Internal,Microsoft.NETCore.App.Ref
 From Version 6.0.301-rtm.22254.17 -> To Version 6.0.400-preview.22302.1 (parent: Microsoft.Android.Sdk.Windows

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Matthew Leibowitz <mattleibow@live.com>
Co-authored-by: Jay Cho <chojoong@gmail.com>
Co-authored-by: Jonathan Peppers <jonathan.peppers@microsoft.com>

rmarinho committed:
Bump iOS versions (#7741)

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/Microsoft.Maui.Graphics build 6.0.400+0-sha.4e9dcc8-azdo.6240183 (#7740)

Microsoft.Maui.Graphics
 From Version 6.0.400-preview.1.1367 -> To Version 6.0.400

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

Redth committed:
More info on templates (#7430)

* More info on templates

- Versioned identity
- Framework 'choice' parameter
- precedence

Related to #7429

* Add more info to item templates

* Remove framework from item templates

* Make template pack id include dotnet version

* Make bigger gap in precedence values

* Update WorkloadManifest.in.json

Co-authored-by: Matthew Leibowitz <mattleibow@live.com>

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/xamarin/xamarin-android build 6.0.4xx-052bfb661739066b8598f5b0b066cee5a99a1e15-1 (#7745)

Microsoft.Android.Sdk.Windows
 From Version 32.0.414 -> To Version 32.0.415

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump Microsoft.CodeAnalysis.NetAnalyzers (#7731)

Bumps [Microsoft.CodeAnalysis.NetAnalyzers](https://github.com/dotnet/roslyn-analyzers) from 7.0.0-preview1.22301.1 to 7.0.0-preview1.22302.1.
- [Release notes](https://github.com/dotnet/roslyn-analyzers/releases)
- [Changelog](https://github.com/dotnet/roslyn-analyzers/blob/main/PostReleaseActivities.md)
- [Commits](https://github.com/dotnet/roslyn-analyzers/commits)

---
updated-dependencies:
- dependency-name: Microsoft.CodeAnalysis.NetAnalyzers
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

phenning committed:
Update ide.host.json schema. (#7736)

Update schema from http://json.schemastore.org/vs-2017.3.host
to http://json.schemastore.org/ide.host

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/xharness build 20220605.1 (#7787)

Microsoft.DotNet.XHarness.CLI , Microsoft.DotNet.XHarness.TestRunners.Common , Microsoft.DotNet.XHarness.TestRunners.Xunit
 From Version 1.0.0-prerelease.22276.1 -> To Version 1.0.0-prerelease.22305.1

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

rmarinho committed:
[housekeeping] Fix signing and insertion for net6 branch (#7791)

* [sign/insertion] Try sign and insert net6 also

* Fix branch name

rookiejava committed:
[Tizen] Fix RendererToHandlerShim.Tizen (#7622)

rmarinho committed:
[Android,iOS] Fix Selection on CollectionView and VSM color changes for Layouts (#7689)

* [Android] Fix clear selection and set correct selected items

* Cleanup

* [iOS,Android] Fix setting color to null on Layouts

* [Tests] Ignore test that fails sometimes WindowHasReasonableDisplayDensity

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/Microsoft.Maui.Graphics build 6.0.403+3-sha.691c654-azdo.6247450 (#7795)

Microsoft.Maui.Graphics
 From Version 6.0.400 -> To Version 6.0.403

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

mattleibow committed:
Fix TemplatedView (#7771)

* Make TemplatedView work as IContentView
* Add tests
* Added TemplatedView control sample

Co-authored-by: Javier Suárez Ruiz <javiersuarezruiz@hotmail.com>

PureWeen committed:
Return VER for PlatformView (#7509)

* Return VER for PlatformView

- If the VisualElementRenderer doesn't set an internal control then just return the VER for the IElementHandler.PlatformView call

* - fix measure calls

dependabot[bot] committed:
Bump Microsoft.WindowsAppSDK from 1.0.3 to 1.1.0 (#7788)

* Bump Microsoft.CodeAnalysis.NetAnalyzers (#7731)

Bumps [Microsoft.CodeAnalysis.NetAnalyzers](https://github.com/dotnet/roslyn-analyzers) from 7.0.0-preview1.22301.1 to 7.0.0-preview1.22302.1.
- [Release notes](https://github.com/dotnet/roslyn-analyzers/releases)
- [Changelog](https://github.com/dotnet/roslyn-analyzers/blob/main/PostReleaseActivities.md)
- [Commits](https://github.com/dotnet/roslyn-analyzers/commits)

---
updated-dependencies:
- dependency-name: Microsoft.CodeAnalysis.NetAnalyzers
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump Microsoft.WindowsAppSDK from 1.0.3 to 1.1.0

Bumps [Microsoft.WindowsAppSDK](https://github.com/microsoft/windowsappsdk) from 1.0.3 to 1.1.0.
- [Release notes](https://github.com/microsoft/windowsappsdk/releases)
- [Commits](https://github.com/microsoft/windowsappsdk/compare/v1.0.3...v1.1.0)

---
updated-dependencies:
- dependency-name: Microsoft.WindowsAppSDK
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update Directory.Build.targets

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Matthew Leibowitz <mattleibow@live.com>

PureWeen committed:
Better handle disappearing content with ViewPager2 (#7425)

* Better handle disappearing content with ViewPager2

* - fix comments and add exist conditions

MackinnonBuck committed:
Improved tab navigation support (#7742)

mattleibow committed:
Use the new layout system in ContentPresenter (#7807)

* Use the new layout system in ContentPresenter

Fixes #7805

* Update PublicAPI.Unshipped.txt

* Update PublicAPI.Unshipped.txt

* Update src/Controls/src/Core/PublicAPI/net6.0-maccatalyst/PublicAPI.Unshipped.txt

* Update src/Controls/src/Core/PublicAPI/net6.0-tizen/PublicAPI.Unshipped.txt

* Update src/Controls/src/Core/PublicAPI/net6.0/PublicAPI.Unshipped.txt

* Update src/Controls/src/Core/PublicAPI/net6.0-windows10.0.19041/PublicAPI.Unshipped.txt

* Update src/Controls/src/Core/PublicAPI/netstandard2.1/PublicAPI.Unshipped.txt

* Update src/Controls/src/Core/PublicAPI/netstandard2.0/PublicAPI.Unshipped.txt

* Update PublicAPI.Unshipped.txt

* Update PublicAPI.Unshipped.txt

mattleibow committed:
Use the SDK band in the rollback files (#7828)

Redth committed:
Stable version branding

mattleibow committed:
New version is now 6.0.5xx

mattleibow committed:
Merge branch 'net6.0'

github-actions[bot] committed:
[create-pull-request] automated change (#7833)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

jsuarezruiz committed:
Fix Shell usage on Catalyst (#7842)

digitaldirk committed:
Update readme to include latest release blogpost (#7831)

Added the blogpost: "Introducing .NET MAUI – One Codebase, Many Platforms" (https://devblogs.microsoft.com/dotnet/introducing-dotnet-maui-one-codebase-many-platforms/) to the readme under "Current News".

If this would be better for an official team member to do, please go ahead : ). I just noticed that the readme was slightly out of date and wanted to fix/bring attention to it.

Thank you

dotnet-maestro[bot] committed:
[net6.0] Update dependencies from xamarin/xamarin-android (#7804)

* Update dependencies from https://github.com/xamarin/xamarin-android build 6.0.4xx-53941c5392925c3c077841b1c2f6827560897209-1

Microsoft.Android.Sdk.Windows
 From Version 32.0.415 -> To Version 32.0.416

Dependency coherency updates

Microsoft.Dotnet.Sdk.Internal
 From Version 6.0.400-preview.22302.1 -> To Version 6.0.400-preview.22306.2 (parent: Microsoft.Android.Sdk.Windows

* Update dependencies from https://github.com/xamarin/xamarin-android build 6.0.4xx-9460d97693c0f05821f90d07fb0b4289e8e9a9a3-1

Microsoft.Android.Sdk.Windows
 From Version 32.0.415 -> To Version 32.0.417

Dependency coherency updates

Microsoft.Dotnet.Sdk.Internal
 From Version 6.0.400-preview.22302.1 -> To Version 6.0.400-preview.22306.7 (parent: Microsoft.Android.Sdk.Windows

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

rachelkang committed:
Fix SearchBar semantics (#7802)

* Fix by updating resource id on android

* Refactor iOS GetSearchTextField code

* Fix iOS SearchView hint & description

* Remove redundant code

* Simplify code a bit more

* Fix failing tests

github-actions[bot] committed:
[create-pull-request] automated change (#7873)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump Xamarin.AndroidX.Palette from 1.0.0.13 to 1.0.0.14 (#7846)

Bumps [Xamarin.AndroidX.Palette](https://github.com/xamarin/AndroidX) from 1.0.0.13 to 1.0.0.14.
- [Release notes](https://github.com/xamarin/AndroidX/releases)
- [Commits](https://github.com/xamarin/AndroidX/commits)

---
updated-dependencies:
- dependency-name: Xamarin.AndroidX.Palette
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump Xamarin.Google.Crypto.Tink.Android from 1.6.1.6 to 1.6.1.7 (#7845)

Bumps [Xamarin.Google.Crypto.Tink.Android](https://github.com/xamarin/AndroidX) from 1.6.1.6 to 1.6.1.7.
- [Release notes](https://github.com/xamarin/AndroidX/releases)
- [Commits](https://github.com/xamarin/AndroidX/commits)

---
updated-dependencies:
- dependency-name: Xamarin.Google.Crypto.Tink.Android
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump Xamarin.AndroidX.Window.WindowJava from 1.0.0.8 to 1.0.0.9 (#7849)

Bumps [Xamarin.AndroidX.Window.WindowJava](https://github.com/xamarin/AndroidX) from 1.0.0.8 to 1.0.0.9.
- [Release notes](https://github.com/xamarin/AndroidX/releases)
- [Commits](https://github.com/xamarin/AndroidX/commits)

---
updated-dependencies:
- dependency-name: Xamarin.AndroidX.Window.WindowJava
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump Xamarin.AndroidX.MediaRouter from 1.3.0 to 1.3.0.1 (#7847)

Bumps [Xamarin.AndroidX.MediaRouter](https://github.com/xamarin/AndroidX) from 1.3.0 to 1.3.0.1.
- [Release notes](https://github.com/xamarin/AndroidX/releases)
- [Commits](https://github.com/xamarin/AndroidX/commits)

---
updated-dependencies:
- dependency-name: Xamarin.AndroidX.MediaRouter
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Redth committed:
Merge branch 'net7.0' into backport/main-net7.0

jonathanpeppers commented on GitInfo.txt:
```diff
@@ -1 +1 @@
-6.0.500-preview.1
+7.0.0-preview.1
```
Maybe doesn't need to happen here, but I already put Android as `-preview.6` because we're on .NET 7 Preview 6.

jonathanpeppers reviewed (approved):
Looks good 👍 , I would use a merge commit when you merge this (don't squash).

Redth committed:
Fix tizen version

Redth committed:
Bump preview version to match sdk

hartez committed:
Fill ScrollView when content is smaller than viewport (#7514)

* Clean up

* Enable FillViewport for Android ScrollView; leave padding/margin content
control always active for Android ScrollView
Fixes #6058 for Android

* Force ScrollView content to be at least the size of the ScrollView itself
Fixes #6058 for iOS

* Verify and remeasure Grid if necessary
Fixes #6058 for Windows

* Ensure ContentSize is set on all platforms

* Move ScrollView content size tests to DeviceTests; make the iOS ScrollView
implementation device testable (and rootable); implement AttachAndRun for iOS

* Make VerticalTextAlignment work for MauiLabel; remove misarrangement of Labels
with backgrounds on iOS;

* Reinstate unused version of method, make real version internal

* Remove unused override

* API changes

* Update overrides for Windows

* Fix invalid structure assumptions in scrollview tests

* Fix 'working on this too late at night' errors

* Revert samples changes

* Better error messages for failing scrollview tests

* Add tolerance to account for rounding

* Better debugging info for iOS AttachAndRun

* Better debugging info for iOS AttachAndRun

* Honestly, we'll take any view

* Grasping at straws now

* Just use whatever view we've got

* Again

jsuarezruiz commented:
/azp run

mattleibow committed:
Move the PublicAPI into unversioned files (#7931)

mattleibow committed:
Merge remote-tracking branch 'origin/net6.0'

# Conflicts:
#	src/Core/src/PublicAPI/net6.0-windows10.0.20348/PublicAPI.Shipped.txt

mattleibow committed:
Merge branch 'main' into backport/main-net7.0

mattleibow committed:
Merge branch 'net7.0' into backport/main-net7.0

mattleibow committed:
Test disabling the public api analyzers

mattleibow committed:
try without Tizen

mattleibow commented:
Having issues with the Public API checkers and one delegate in Tizen.

mattleibow committed:
Enable analyzers again

jsuarezruiz committed:
Fix exception using FlyoutBackgroundImage on iOS (#7879)

jfversluis committed:
Update MobileAuthController.cs (#7983)

TanayParikh committed:
Fix `maui-blazor` indentation (#7976)

Tabs -> spaces consistency.

Fixes: https://github.com/dotnet/maui/issues/7934

TanayParikh committed:
Add default `Name=blazorWebView` to MAUI MainPage.xaml (#8007)

Any harm in having this there by default?

Currently our instructions are a bit ambiguous around this area: https://docs.microsoft.com/en-us/aspnet/core/blazor/hybrid/routing?view=aspnetcore-6.0#net-maui

TanayParikh committed:
Remove RC2 & Preview 14 Development Workload Installation Instructions (#8006)

Do we still need to keep these?

jfversluis committed:
Update bug-report.yml

StephaneDelcroix committed:
[C] fix BP declaration for Border (#8047)

- fixes #7744

jonathanpeppers committed:
[android] reduce JNI calls during layout measurement (#8034)

Context: https://github.com/unoplatform/performance/tree/master/src/dopes/DopeTestMaui

Building upon #7996, #8001, and #8033, I noticed while profiling the
sample app putting N Label on the screen:

    783.92ms (6.2%) microsoft.maui!Microsoft.Maui.ViewHandlerExtensions.GetDesiredSizeFromHandler(Microsoft.Maui.IViewHandler,double,double)

So around %6 of the time spend just measuring.

Looking through the call stack, I can see 3 JNI calls happening:

    932.51ms (7.4%)  mono.android!Android.Views.View.Measure(int,int)
    115.53ms (0.91%) mono.android!Android.Views.View.get_MeasuredWidth()
     96.97ms (0.77%) mono.android!Android.Views.View.get_MeasuredHeight()

So, we could write a Java method that calls all three of these and
somehow returns the `MeasuredWidth` and `MeasuredHeight`. After a
little research, it seemed the best approach here was to "pack" two
integers into a `long`. If we tried to return some Java object
instead, then we'd have the same number of JNI calls to get the
integers out.

I found a couple links describing how to "pack" an `int` into a `long`:

* Java: https://stackoverflow.com/a/12772968
* C#: https://stackoverflow.com/a/827267

Which after some testing, arrives at:

    public static long measureAndGetWidthAndHeight(View view, int widthMeasureSpec, int heightMeasureSpec) {
        view.measure(widthMeasureSpec, heightMeasureSpec);
        int width = view.getMeasuredWidth();
        int height = view.getMeasuredHeight();
        return ((long)width << 32) | (height & 0xffffffffL);
    }

Unpacked in C# such as:

    var packed = PlatformInterop.MeasureAndGetWidthAndHeight(platformView, widthSpec, heightSpec);
    var measuredWidth = (int)(packed >> 32);
    var measuredHeight = (int)(packed & 0xffffffffL);

Reducing 3 JNI calls in every `View`'s layout to 1.

~~ Results ~~

A `Release` build on a Pixel 5 device, I was getting:

    Before:  91.94 Dopes/s
    After:  102.45 Dopes/s

After profiling again, it drops the % time spent in
`GetDesiredSizeFromHandler`:

    528.96ms (4.5%) microsoft.maui!Microsoft.Maui.ViewHandlerExtensions.GetDesiredSizeFromHandler

So the added math is negligible compared to reduced JNI calls.

mattleibow committed:
Revert "try without Tizen"

This reverts commit 60ed1d9e88f823c9c0319adad413b76cf6756981.

mattleibow committed:
Merge remote-tracking branch 'origin/main' into backport/main-net7.0

mattleibow committed:
Fixed the issues!

mattleibow committed:
More fixes for other dlls

mattleibow committed:
Fix this potential issue
